geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseArrowTest.scala

```diff
@@ -242,6 +242,43 @@ class HBaseArrowTest extends HBaseTest with LazyLogging  {
         }
       }
     }
+    "work with partitioned tables and arrow coprocessor disabled" in {
+      val sft = SimpleFeatureTypes.createType("arrow-pds",
+        "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;" +
+          "geomesa.table.partition=time,geomesa.z3.interval=day,geomesa.indices.enabled=z3")
+      ds.createSchema(sft)
+      val features = Seq.tabulate(8) { i =>
+        ScalaSimpleFeature.create(sft, s"$i", s"name${i % 2}", s"${i % 5}", s"2017-02-0${i+1}T00:00:01.000Z", s"POINT(40 6$i)")
+      }
+      WithClose(ds.getFeatureWriterAppend(sft.getTypeName, Transaction.AUTO_COMMIT)) { writer =>
+        features.foreach { f =>
+          FeatureUtils.copyToWriter(writer, f, useProvidedFid = true)
+          writer.write()
+        }
+      }
+      HBaseDataStoreFactory.RemoteArrowProperty.threadLocalValue.set("false")
+      try {
+        val query = new Query(sft.getTypeName, Filter.INCLUDE, Array("name", "dtg", "geom"))
+        query.getHints.put(QueryHints.ARROW_ENCODE, java.lang.Boolean.TRUE)
+        query.getHints.put(QueryHints.ARROW_DICTIONARY_FIELDS, "name")
+        query.getHints.put(QueryHints.ARROW_SORT_FIELD, "dtg")
+        query.getHints.put(QueryHints.ARROW_INCLUDE_FID, java.lang.Boolean.FALSE)
+        val results = SelfClosingIterator(ds.getFeatureReader(query, Transaction.AUTO_COMMIT))
+        val out = new ByteArrayOutputStream
+        results.foreach(sf => out.write(sf.getAttribute(0).asInstanceOf[Array[Byte]]))
+        def in() = new ByteArrayInputStream(out.toByteArray)
+        WithClose(ArrowAllocator("hbase-batch-pds-test")) { allocator =>
+          WithClose(SimpleFeatureArrowFileReader.caching(in())(allocator)) { reader =>
+            SelfClosingIterator(reader.features()).map(f => f.getID -> f.getAttributes.asScala).toList mustEqual
+                features.map(f => f.getID -> query.getPropertyNames.map(f.getAttribute).toSeq)
+            // ensure the reduce step worked correctly and there is only 1 logical file
+            reader.vectors must haveLength(1)
```

Comment:
@jnh5y i seem to recall we were looking for a way to validate how many logical arrow files are in an arrow result - for future reference, this is how you can do it.

Reply:
Yeah, I think that was in a discussion with @aheyne about an issue he was hitting.